app/controllers/publications_controller.rb

```diff
@@ -3,8 +3,10 @@
 ##
 # Controller to handle actions related to a publication
 class PublicationsController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:claim]
+  before_action :login_claim_link_user, only: [:claim]
```

Comment:
It looks like you could combine this method with the record_by_hashed_email method and do the whole thing in one shot

```
@author_publication = AuthorPublication.includes(:publication)
                                       .where(claim_link: params[:claim_link])
                                       .first
@user = @author_publication.user
sign_in(@user)
@publication = @author_publication.publication
```